### ISSUE TITLE
[JENKINS-12875] "No valid crumb was included in the request" errors all around

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-23#1149]
<p>After upgrading Jenkins from 1.448 to 1.451 (with yum, upgraded plugins too) several parts of Jenkins have become unusable with error "No valid crumb was included in the request":</p>

<h4><a name="Consoleoutput"></a>Console output</h4>

<ul>
	<li>When building, the log is unavailable, see <span class="nobr"><a href="https://issues.jenkins.io/secure/attachment/21553/21553_console_log.png" title="console_log.png attached to JENKINS-12875">console_log.png<sup><img class="rendericon" src="https://issues.jenkins.io/images/icons/link_attachment_7.gif" height="7" width="7" align="absmiddle" alt="" border="0"/></sup></a></span></li>
	<li>Once the build is done, the complete log is available, but the same error appears on the side <span class="nobr"><a href="https://issues.jenkins.io/secure/attachment/21552/21552_console_side.png" title="console_side.png attached to JENKINS-12875">console_side.png<sup><img class="rendericon" src="https://issues.jenkins.io/images/icons/link_attachment_7.gif" height="7" width="7" align="absmiddle" alt="" border="0"/></sup></a></span></li>
</ul>

<h4><a name="Buildconfiguration"></a>Build configuration</h4>

<ul>
	<li>Builds cannot be launched anymore from the build page (The firebug error console displays the "No valid crumb was included in the request" error). They still can be launched from the dashboard though.</li>
	<li>Cannot add build steps (when choosing a build step type an error is reported in background), see <span class="nobr"><a href="https://issues.jenkins.io/secure/attachment/21556/21556_add_build_step.png" title="add_build_step.png attached to JENKINS-12875">add_build_step.png<sup><img class="rendericon" src="https://issues.jenkins.io/images/icons/link_attachment_7.gif" height="7" width="7" align="absmiddle" alt="" border="0"/></sup></a></span></li>
</ul>

<h4><a name="ManageJenkinsconfiguration"></a>Manage Jenkins configuration</h4>

<ul>
	<li>The page loads only partially (first half is permanently greyed out, see <span class="nobr"><a href="https://issues.jenkins.io/secure/attachment/21555/21555_configuration.png" title="configuration.png attached to JENKINS-12875">configuration.png<sup><img class="rendericon" src="https://issues.jenkins.io/images/icons/link_attachment_7.gif" height="7" width="7" align="absmiddle" alt="" border="0"/></sup></a></span>). The firebug error console displays the "No valid crumb was included in the request" error.</li>
</ul>

<h4><a name="ManageJenkinsplugins"></a>Manage Jenkins plugins</h4>

<ul>
	<li>Cannot upgrade nor downgrade, see error shown <span class="nobr"><a href="https://issues.jenkins.io/secure/attachment/21554/21554_plugins.png" title="plugins.png attached to JENKINS-12875">plugins.png<sup><img class="rendericon" src="https://issues.jenkins.io/images/icons/link_attachment_7.gif" height="7" width="7" align="absmiddle" alt="" border="0"/></sup></a></span></li>
</ul>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=naloy">naloy</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-12875" target="_blank">"No valid crumb was included in the request" errors all around</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Critical
<li><b>component(s)</b>: core
<li><b>label(s)</b>: 2.0, 2.0-planned, crumb
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2016-04-01T23:57:26+00:00
<li><b>votes</b>: 20
<li><b>watchers</b>: 43
<li><b>imported</b>: 2025-11-23
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>After upgrading Jenkins from 1.448 to 1.451 (with yum, upgraded plugins too) several parts of Jenkins have become unusable with error "No valid crumb was included in the request":</p>

<h4><a name="Consoleoutput"></a>Console output</h4>

<ul>
	<li>When building, the log is unavailable, see <span class="nobr"><a href="https://issues.jenkins.io/secure/attachment/21553/21553_console_log.png" title="console_log.png attached to JENKINS-12875">console_log.png<sup><img class="rendericon" src="https://issues.jenkins.io/images/icons/link_attachment_7.gif" height="7" width="7" align="absmiddle" alt="" border="0"/></sup></a></span></li>
	<li>Once the build is done, the complete log is available, but the same error appears on the side <span class="nobr"><a href="https://issues.jenkins.io/secure/attachment/21552/21552_console_side.png" title="console_side.png attached to JENKINS-12875">console_side.png<sup><img class="rendericon" src="https://issues.jenkins.io/images/icons/link_attachment_7.gif" height="7" width="7" align="absmiddle" alt="" border="0"/></sup></a></span></li>
</ul>

<h4><a name="Buildconfiguration"></a>Build configuration</h4>

<ul>
	<li>Builds cannot be launched anymore from the build page (The firebug error console displays the "No valid crumb was included in the request" error). They still can be launched from the dashboard though.</li>
	<li>Cannot add build steps (when choosing a build step type an error is reported in background), see <span class="nobr"><a href="https://issues.jenkins.io/secure/attachment/21556/21556_add_build_step.png" title="add_build_step.png attached to JENKINS-12875">add_build_step.png<sup><img class="rendericon" src="https://issues.jenkins.io/images/icons/link_attachment_7.gif" height="7" width="7" align="absmiddle" alt="" border="0"/></sup></a></span></li>
</ul>

<h4><a name="ManageJenkinsconfiguration"></a>Manage Jenkins configuration</h4>

<ul>
	<li>The page loads only partially (first half is permanently greyed out, see <span class="nobr"><a href="https://issues.jenkins.io/secure/attachment/21555/21555_configuration.png" title="configuration.png attached to JENKINS-12875">configuration.png<sup><img class="rendericon" src="https://issues.jenkins.io/images/icons/link_attachment_7.gif" height="7" width="7" align="absmiddle" alt="" border="0"/></sup></a></span>). The firebug error console displays the "No valid crumb was included in the request" error.</li>
</ul>

<h4><a name="ManageJenkinsplugins"></a>Manage Jenkins plugins</h4>

<ul>
	<li>Cannot upgrade nor downgrade, see error shown <span class="nobr"><a href="https://issues.jenkins.io/secure/attachment/21554/21554_plugins.png" title="plugins.png attached to JENKINS-12875">plugins.png<sup><img class="rendericon" src="https://issues.jenkins.io/images/icons/link_attachment_7.gif" height="7" width="7" align="absmiddle" alt="" border="0"/></sup></a></span></li>
</ul>
</pre>
</details>
</details>
<ul><li><i>environment</i>: <code>CentOS release 5.5 (Final) 64b</code></li></ul>
<details><summary><i>6 attachments</i></summary>

- [Screen Shot 2014-03-12 at 10.43.58 AM.png](https://issues.jenkins.io/secure/attachment/25542/Screen%20Shot%202014-03-12%20at%2010.43.58%20AM.png)
  > ![Screen Shot 2014-03-12 at 10.43.58 AM.png](https://issues.jenkins.io/secure/attachment/25542/Screen%20Shot%202014-03-12%20at%2010.43.58%20AM.png)
- [add_build_step.png](https://issues.jenkins.io/secure/attachment/21556/add_build_step.png)
  > ![add_build_step.png](https://issues.jenkins.io/secure/attachment/21556/add_build_step.png)
- [configuration.png](https://issues.jenkins.io/secure/attachment/21555/configuration.png)
  > ![configuration.png](https://issues.jenkins.io/secure/attachment/21555/configuration.png)
- [console_log.png](https://issues.jenkins.io/secure/attachment/21553/console_log.png)
  > ![console_log.png](https://issues.jenkins.io/secure/attachment/21553/console_log.png)
- [console_side.png](https://issues.jenkins.io/secure/attachment/21552/console_side.png)
  > ![console_side.png](https://issues.jenkins.io/secure/attachment/21552/console_side.png)
- [plugins.png](https://issues.jenkins.io/secure/attachment/21554/plugins.png)
  > ![plugins.png](https://issues.jenkins.io/secure/attachment/21554/plugins.png)
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>naloy:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-12875?focusedId=159562&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-159562">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>After yet another restart things seems to be back to normal (???!!)</p>
</pre>
</details>

</ul>
</details>

<p>After yet another restart things seems to be back to normal (???!!)</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/valentine.jpg" title="valentine's avatar" /> valentine:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-12875?focusedId=159565&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-159565">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Not completely normal. The loading thing in the management section is still happening </p>
</pre>
</details>

</ul>
</details>

<p>Not completely normal. The loading thing in the management section is still happening </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>naloy:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-12875?focusedId=159566&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-159566">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>In the Jenkins configuration page:</p>

<p>Error: Form.findMatchingInput is not a function
Source File: <a href="http://jenkins.rightnow.com:8080/static/6dcd23a9/scripts/hudson-behavior.js" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins.rightnow.com:8080/static/6dcd23a9/scripts/hudson-behavior.js</a>
Line: 338</p>
</pre>
</details>

</ul>
</details>

<p>In the Jenkins configuration page:</p>

<p>Error: Form.findMatchingInput is not a function<br/>
Source File: <a href="http://jenkins.rightnow.com:8080/static/6dcd23a9/scripts/hudson-behavior.js" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins.rightnow.com:8080/static/6dcd23a9/scripts/hudson-behavior.js</a><br/>
Line: 338</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>windyroad:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-12875?focusedId=168028&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-168028">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I'm not as blocked as the reporting user, but I'm having the same problem (no valid crumb..). In my case it only seems to affect the console output, though the error can be seen for "ajaxBuildQueue" and "ajaxExecuters" on the config page, the home page (and I'm assuming others).</p>

<p>What's interesting is that I only get this problem through a reverse proxy. <a href="http://ve.mbqblqmj.vesrv.com:8080/" class="external-link" target="_blank" rel="nofollow noopener">http://ve.mbqblqmj.vesrv.com:8080/</a> works fine, but <a href="http://ci.windyroad.org/" class="external-link" target="_blank" rel="nofollow noopener">http://ci.windyroad.org/</a> (which is RPed to  <a href="http://ve.mbqblqmj.vesrv.com:8080/" class="external-link" target="_blank" rel="nofollow noopener">http://ve.mbqblqmj.vesrv.com:8080/</a>) does not.</p>
</pre>
</details>

</ul>
</details>

<p>I'm not as blocked as the reporting user, but I'm having the same problem (no valid crumb..). In my case it only seems to affect the console output, though the error can be seen for "ajaxBuildQueue" and "ajaxExecuters" on the config page, the home page (and I'm assuming others).</p>

<p>What's interesting is that I only get this problem through a reverse proxy. <a href="http://ve.mbqblqmj.vesrv.com:8080/" class="external-link" target="_blank" rel="nofollow noopener">http://ve.mbqblqmj.vesrv.com:8080/</a> works fine, but <a href="http://ci.windyroad.org/" class="external-link" target="_blank" rel="nofollow noopener">http://ci.windyroad.org/</a> (which is RPed to  <a href="http://ve.mbqblqmj.vesrv.com:8080/" class="external-link" target="_blank" rel="nofollow noopener">http://ve.mbqblqmj.vesrv.com:8080/</a>) does not.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>windyroad:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-12875?focusedId=168029&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-168029">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I've offered  USD100.00 at FreedomSponsors to have this fixed (<a href="http://www.freedomsponsors.org/core/issue/31/no-valid-crumb-was-included-in-the-request-errors-all-around" class="external-link" target="_blank" rel="nofollow noopener">http://www.freedomsponsors.org/core/issue/31/no-valid-crumb-was-included-in-the-request-errors-all-around</a>).
Once you solve it (according to the acceptance criteria described there), just create a FreedomSponsors account and mark it as resolved (oh, you'll need a Paypal account too)
I'll then check it out and will gladly pay.</p>
</pre>
</details>

</ul>
</details>

<p>I've offered  USD100.00 at FreedomSponsors to have this fixed (<a href="http://www.freedomsponsors.org/core/issue/31/no-valid-crumb-was-included-in-the-request-errors-all-around" class="external-link" target="_blank" rel="nofollow noopener">http://www.freedomsponsors.org/core/issue/31/no-valid-crumb-was-included-in-the-request-errors-all-around</a>).<br/>
Once you solve it (according to the acceptance criteria described there), just create a FreedomSponsors account and mark it as resolved (oh, you'll need a Paypal account too)<br/>
I'll then check it out and will gladly pay.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>onemanbucket:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-12875?focusedId=169308&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-169308">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Actually, it seems that you can set the header name with a system property to jenkins (hudson.security.csrf.requestfield). So another workaround could be to start jenkins with -Dhudson.security.csrf.requestfield=crumb</p>
</pre>
</details>

</ul>
</details>

<p>Actually, it seems that you can set the header name with a system property to jenkins (hudson.security.csrf.requestfield). So another workaround could be to start jenkins with -Dhudson.security.csrf.requestfield=crumb</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>algob:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-12875?focusedId=174479&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-174479">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I've same issue, if I try to disable a project (use Jenkins v1.502):
   <a href="http://rhodan:8080/jenkins/job/myproject_3562/disable" class="external-link" target="_blank" rel="nofollow noopener">http://rhodan:8080/jenkins/job/myproject_3562/disable</a></p>

<p>It looks like, I would have the same issue, if I would try to enable it, if it is disabled: <a href="https://issues.jenkins.io/browse/JENKINS-14501" title="Enable Job link fails with &quot;No valid crumb was included in the request&quot;" class="issue-link" data-issue-key="JENKINS-14501"><del>JENKINS-14501</del></a></p>

<p>I tried to added these line to /opt/tomcat/bin/catalina.sh, but it didn't solve the issue:
   &#8212; SNIP &#8212;
JAVA_OPTS="$JAVA_OPTS -Dhudson.security.csrf.requestfield=crumb"
   &#8212; SNAP &#8212;</p>
</pre>
</details>

</ul>
</details>

<p>I've same issue, if I try to disable a project (use Jenkins v1.502):<br/>
   <a href="http://rhodan:8080/jenkins/job/myproject_3562/disable" class="external-link" target="_blank" rel="nofollow noopener">http://rhodan:8080/jenkins/job/myproject_3562/disable</a></p>

<p>It looks like, I would have the same issue, if I would try to enable it, if it is disabled: <a href="https://issues.jenkins.io/browse/JENKINS-14501" title="Enable Job link fails with "No valid crumb was included in the request"" class="issue-link" data-issue-key="JENKINS-14501"><del>JENKINS-14501</del></a></p>

<p>I tried to added these line to /opt/tomcat/bin/catalina.sh, but it didn't solve the issue:<br/>
   &#8212; SNIP &#8212;<br/>
JAVA_OPTS="$JAVA_OPTS -Dhudson.security.csrf.requestfield=crumb"<br/>
   &#8212; SNAP &#8212;</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ozawa.jpg" title="ozawa's avatar" /> ozawa:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-12875?focusedId=179142&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-179142">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Today, I've had also the same issue. Delete jobs or clone from github hasn't worked correctly.
This problem is caused after upgrading 1.515.</p>
</pre>
</details>

</ul>
</details>

<p>Today, I've had also the same issue. Delete jobs or clone from github hasn't worked correctly.<br/>
This problem is caused after upgrading 1.515.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-12875?focusedId=254374&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-254374">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jknurek" class="user-hover" rel="jknurek">jknurek</a> That's unrelated to this issue and looks like a bug in the plugin that cannot handle the CSRF protection option (around since 2010 or so) being enabled.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jknurek" class="user-hover" rel="jknurek">jknurek</a> That's unrelated to this issue and looks like a bug in the plugin that cannot handle the CSRF protection option (around since 2010 or so) being enabled.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jknurek.jpg" title="jknurek's avatar" /> jknurek:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-12875?focusedId=254597&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-254597">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>My first thought was that it was a bug specific to the Build Monitor View plugin, but when I run Jenkins 1.651.1, the same configuration works ok. I'm only getting this error with the 2.0-rc-1. 
(but I'm not sure what the underlying issue is in this case, so it is likely a separate issue)</p>
</pre>
</details>

</ul>
</details>

<p>My first thought was that it was a bug specific to the Build Monitor View plugin, but when I run Jenkins 1.651.1, the same configuration works ok. I'm only getting this error with the 2.0-rc-1. <br/>
(but I'm not sure what the underlying issue is in this case, so it is likely a separate issue)</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/janek.png" title="janek's avatar" /> janek:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-12875?focusedId=255957&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-255957">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck" class="user-hover" rel="danielbeck">danielbeck</a> Build Monitor has supported CSRF protection since its early days <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/> </p>

<p>The problem is caused by change <a href="https://github.com/jenkinsci/jenkins/compare/4649e04cbe58...a87c38c5052c" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/compare/4649e04cbe58...a87c38c5052c</a> , which replaced the hard-coded '.crumb' header with 'Jenkins-Crumb', that's why they no longer match.</p>

<p>I can work around that in Build Monitor.</p>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jknurek" class="user-hover" rel="jknurek">jknurek</a> you can watch the progress of this issue on github at <a href="https://github.com/jan-molak/jenkins-build-monitor-plugin/issues/215" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jan-molak/jenkins-build-monitor-plugin/issues/215</a>, which is the official issue tracker for Build Monitor.</p>

<p>Thanks!
Jan</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck" class="user-hover" rel="danielbeck">danielbeck</a> Build Monitor has supported CSRF protection since its early days <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/> </p>

<p>The problem is caused by change <a href="https://github.com/jenkinsci/jenkins/compare/4649e04cbe58...a87c38c5052c" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/compare/4649e04cbe58...a87c38c5052c</a> , which replaced the hard-coded '.crumb' header with 'Jenkins-Crumb', that's why they no longer match.</p>

<p>I can work around that in Build Monitor.</p>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jknurek" class="user-hover" rel="jknurek">jknurek</a> you can watch the progress of this issue on github at <a href="https://github.com/jan-molak/jenkins-build-monitor-plugin/issues/215" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jan-molak/jenkins-build-monitor-plugin/issues/215</a>, which is the official issue tracker for Build Monitor.</p>

<p>Thanks!<br/>
Jan</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-12875?focusedId=255963&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-255963">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=janek" class="user-hover" rel="janek">janek</a></p>

<blockquote><p>which replaced the hard-coded '.crumb' header with 'Jenkins-Crumb', that's why they no longer match.</p></blockquote>

<p><a href="https://wiki.jenkins-ci.org/display/JENKINS/Features+controlled+by+system+properties" class="external-link" target="_blank" rel="nofollow noopener">The crumb name has been configurable since 1.310</a> (<tt>hudson.security.csrf.requestfield</tt>), and that was necessary for anyone running Jenkins behind nginx, as it dropped the <tt>.crumb</tt> header (<a href="https://issues.jenkins.io/browse/JENKINS-12875" title="&quot;No valid crumb was included in the request&quot; errors all around" class="issue-link" data-issue-key="JENKINS-12875"><del>JENKINS-12875</del></a>). So we renamed the default to something nginx accepts. Maybe it's a good idea to check which crumb name is used by Jenkins in your plugin to adapt to user configuration.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=janek" class="user-hover" rel="janek">janek</a></p>

<blockquote><p>which replaced the hard-coded '.crumb' header with 'Jenkins-Crumb', that's why they no longer match.</p></blockquote>

<p><a href="https://wiki.jenkins-ci.org/display/JENKINS/Features+controlled+by+system+properties" class="external-link" target="_blank" rel="nofollow noopener">The crumb name has been configurable since 1.310</a> (<tt>hudson.security.csrf.requestfield</tt>), and that was necessary for anyone running Jenkins behind nginx, as it dropped the <tt>.crumb</tt> header (<a href="https://issues.jenkins.io/browse/JENKINS-12875" title=""No valid crumb was included in the request" errors all around" class="issue-link" data-issue-key="JENKINS-12875"><del>JENKINS-12875</del></a>). So we renamed the default to something nginx accepts. Maybe it's a good idea to check which crumb name is used by Jenkins in your plugin to adapt to user configuration.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/janek.png" title="janek's avatar" /> janek:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-12875?focusedId=255967&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-255967">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jknurek" class="user-hover" rel="jknurek">jknurek</a> - Fixed in Build Monitor 1.9+build.2016043001​28, which you can <a href="https://smartcode-opensource.ci.cloudbees.com/job/build-monitor/37/artifact/build-monitor-plugin/target/build-monitor-plugin.hpi" class="external-link" target="_blank" rel="nofollow noopener">download here</a> before it hits the official update centre.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jknurek" class="user-hover" rel="jknurek">jknurek</a> - Fixed in Build Monitor 1.9+build.2016043001​28, which you can <a href="https://smartcode-opensource.ci.cloudbees.com/job/build-monitor/37/artifact/build-monitor-plugin/target/build-monitor-plugin.hpi" class="external-link" target="_blank" rel="nofollow noopener">download here</a> before it hits the official update centre.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/janek.png" title="janek's avatar" /> janek:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-12875?focusedId=255969&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-255969">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Awesome, thanks <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck" class="user-hover" rel="danielbeck">danielbeck</a>, that's exactly what I did. The latest Build Monitor picks up the crumb name from CrumbIssuer::getCrumbRequestField so it should be a more appropriate solution <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>
</pre>
</details>

</ul>
</details>

<p>Awesome, thanks <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck" class="user-hover" rel="danielbeck">danielbeck</a>, that's exactly what I did. The latest Build Monitor picks up the crumb name from CrumbIssuer::getCrumbRequestField so it should be a more appropriate solution <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>nvn:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-12875?focusedId=261957&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-261957">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Hi,
I have Jenkins 2.7, I have created the service hook in TFS to trigger the automatic build but while testing the hook I get " No valid crumb was included in the request (403)". I can run the same build from Jenkins dashboard with no issue.
I have also installed the downgrade Build - Plugin-Version: 1.9+build.201606171408 as suggested by Jan Molak above but I am still facing the issue.</p>

<p>P.s - I have installed Jenkins on my local windows 2007 machine.</p>

<p>Thanks,</p>
</pre>
</details>

</ul>
</details>

<p>Hi,<br/>
I have Jenkins 2.7, I have created the service hook in TFS to trigger the automatic build but while testing the hook I get " No valid crumb was included in the request (403)". I can run the same build from Jenkins dashboard with no issue.<br/>
I have also installed the downgrade Build - Plugin-Version: 1.9+build.201606171408 as suggested by Jan Molak above but I am still facing the issue.</p>

<p>P.s - I have installed Jenkins on my local windows 2007 machine.</p>

<p>Thanks,</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/janek.png" title="janek's avatar" /> janek:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-12875?focusedId=261958&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-261958">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Hey Naveen - from what you're saying it seems like the problem is related to Jenkins and not the Build Monitor plugin?
If you think it's related to Build Monitor please feel free to raise a ticket on github at <a href="https://github.com/jan-molak/jenkins-build-monitor-plugin" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jan-molak/jenkins-build-monitor-plugin</a></p>

<p>Thanks,
Jan</p>
</pre>
</details>

</ul>
</details>

<p>Hey Naveen - from what you're saying it seems like the problem is related to Jenkins and not the Build Monitor plugin?<br/>
If you think it's related to Build Monitor please feel free to raise a ticket on github at <a href="https://github.com/jan-molak/jenkins-build-monitor-plugin" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jan-molak/jenkins-build-monitor-plugin</a></p>

<p>Thanks,<br/>
Jan</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>nvn:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-12875?focusedId=261967&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-261967">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Hi Jan, yeah I think the issue is either with Jenkins or may be with Tfs-jenkins plugin, I am not sure. Pls suggest.</p>

<p>Thanks</p>
</pre>
</details>

</ul>
</details>

<p>Hi Jan, yeah I think the issue is either with Jenkins or may be with Tfs-jenkins plugin, I am not sure. Pls suggest.</p>

<p>Thanks</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>12niner:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-12875?focusedId=273684&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-273684">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Having just wasted a few hours debugging why I was seeing this: if you re-install Jenkins you need to clear your cookies or you'll hit this.</p>
</pre>
</details>

</ul>
</details>

<p>Having just wasted a few hours debugging why I was seeing this: if you re-install Jenkins you need to clear your cookies or you'll hit this.</p>
